src/cross_section.F90

```diff
@@ -153,87 +167,119 @@ subroutine calculate_nuclide_xs(i_nuclide, i_sab, E, i_mat, i_nuc_mat, i_log_uni
     nuc => nuclides(i_nuclide)
     mat => materials(i_mat)
 
-    ! Determine index on nuclide energy grid
-    select case (grid_method)
-    case (GRID_MAT_UNION)
+    ! If MP, don't interpolate, it's all already baked in.
+    if( nuc % mp_present .AND. &
+         (E >= nuc % multipole % start_E/1.0D6 .AND.&
+          E <= nuc % multipole % end_E/1.0D6)) then
```

Comment:
For the sake of consistency, I'd suggest using `1.0e6_8`. We don't have any other instances in the code that use the `x.xxDx` notation